src/commands/lightning/preview/app.ts

```diff
@@ -1,41 +1,137 @@
   public static readonly summary = messages.getMessage('summary');
   public static readonly description = messages.getMessage('description');
   public static readonly examples = messages.getMessages('examples');
 
   public static readonly flags = {
     name: Flags.string({
       summary: messages.getMessage('flags.name.summary'),
-      description: messages.getMessage('flags.name.description'),
       char: 'n',
-      required: false,
+    }),
+    'target-org': Flags.requiredOrg({
+      summary: messages.getMessage('flags.target-org.summary'),
+    }),
+    'device-type': Flags.option({
+      summary: messages.getMessage('flags.device-type.summary'),
+      char: 't',
+      options: [Platform.desktop, Platform.ios, Platform.android] as const,
+      default: Platform.desktop,
+    })(),
+    'device-id': Flags.string({
+      summary: messages.getMessage('flags.device-id.summary'),
+      char: 'i',
     }),
   };
 
-  public async run(): Promise<LightningPreviewAppResult> {
+  public async run(): Promise<void> {
     const { flags } = await this.parse(LightningPreviewApp);
+    const logger = await Logger.child(this.ctor.name);
+
+    if (flags['device-type'] === Platform.desktop) {
+      await this.desktopPreview(logger, flags);
+    } else if (flags['device-type'] === Platform.ios) {
+      await this.iosPreview(logger, flags);
+    } else if (flags['device-type'] === Platform.android) {
+      await this.androidPreview(logger, flags);
+    }
+  }
+
+  private async desktopPreview(logger: Logger, flags: LightningPreviewAppFlags): Promise<void> {
+    const appName = flags['name'];
+    const targetOrg = flags['target-org'];
+    const platform = flags['device-type'];
+
+    logger.debug('Determining Local Dev Server url');
+    // todo: figure out how to make the port dynamic instead of hard-coded value here
+    const ldpServerUrl = PreviewUtils.generateWebSocketUrlForLocalDevServer(platform, '8081');
+    logger.debug(`Local Dev Server url is ${ldpServerUrl}`);
+
+    // appPath will resolve to one of the following:
+    //
+    //   lightning/app/<appId> => when the user is targetting a specific LEX app
+    //               lightning => when the user is not targetting a specific LEX app
+    //
+    let appPath = '';
+    if (appName) {
+      logger.debug(`Determining App Id for ${appName}`);
+
+      // The appName is optional but if the user did provide an appName then it must be
+      // a valid one.... meaning that it should resolve to a valid appId.
+      const appId = await OrgUtils.getAppId(targetOrg.getConnection(undefined), appName);
+      if (!appId) {
+        throw new Error(messages.getMessage('error.fetching.app-id', [appName]));
+      }
+
+      logger.debug(`App Id is ${appId}`);
+
+      appPath = `lightning/app/${appId}`;
+    } else {
+      logger.debug('No Lightning Experience application name provided.... using the default app instead.');
```

Comment:
Nit: this is probably a good candidate for `logger.info()` instead. Wouldn't hurt to routinely notify the user of their choice here.

Reply:
Sure... but just FYI `logger.blah` will only output to `~/.sf/log-file` and not to `stdout`. I didn't want to have anything extra printed to `stdout` b/c I was trying to follow Marcelino's doc with the example output.

Reply:
Oh okay. If it's all going to a log file anyway, it's probably less important.